create_functions.sql

```diff
@@ -417,56 +510,170 @@ BEGIN
             acc.slot,
             acc.write_version,
             acc.txn_signature
-        FROM account_audit AS acc
+        FROM public.account_audit AS acc
         WHERE
             acc.slot <= in_slot
             AND acc.pubkey = in_pubkey
-        ORDER BY
-            acc.slot DESC, acc.write_version DESC
+        ORDER BY 
+            acc.pubkey DESC, 
+            acc.write_version DESC 
         LIMIT 1;
 END;
-$get_account_at_slot$ LANGUAGE plpgsql;
+$get_account_at_slot_from_audit$ LANGUAGE plpgsql;
 
 -----------------------------------------------------------------------------------------------------------------------
-
-CREATE OR REPLACE FUNCTION get_recent_blockhashes(start_slot BIGINT, num_hashes INT)
+CREATE OR REPLACE FUNCTION get_account_at_slot(
+    in_pubkey BYTEA,
+    in_slot BIGINT
+)
 
 RETURNS TABLE (
-	slot BIGINT,
-    blockhash VARCHAR(44)
+    pubkey BYTEA,
+	owner BYTEA,
+	lamports BIGINT,
+	executable BOOL,
+	rent_epoch BIGINT,
+	data BYTEA,
+    slot BIGINT,
+    write_version BIGINT,
+    signature BYTEA
 )
 
-AS $get_recent_blockhashes$
+AS $get_account_at_slot$
+
+BEGIN
+    RETURN QUERY
+        WITH results AS (
+            SELECT * FROM get_account_at_slot_from_audit(in_pubkey, in_slot)
```

Comment:
account_audit and older_account can be joned by VIEW.
Will it be useful?
It can simplify stored procedures and functions.

Reply:
[discussed]